Max feature count docs (#1695)
urbica/martin: docs/src/config-file.md

  # Maximum Postgres connections pool size [default: 20]
  pool_size: 20

>   # Limit the number of table geo features included in a tile.

Reviewer: ```suggestion
  # Limit the number of geo features per tile.
```

  #

>   # If the number of features is higher than this, they will not be included in the tile and thus look "cut off"/incomplete.

Reviewer: ```suggestion
  # If the source table has more features than set here, they will not be included in the tile and the result will look "cut off"/incomplete.
```

  # This feature allows to put a maximum latency bound on tiles with extreme amount of detail at the cost of not returning all data.

>   # It is sensible to have such a limit in place if you have user generated/untrusted geodata. (Think: lots of datapoints near null island)

Reviewer: ```suggestion
  # It is sensible to set this limit if you have user generated/untrusted geodata, e.g. a lot of data points at  [Null Island](https://en.wikipedia.org/wiki/Null_Island).
```

Author: lets not have double spaces, otherwise quite reasonable

```suggestion
  # It is sensible to set this limit if you have user generated/untrusted geodata, e.g. a lot of data points at [Null Island](https://en.wikipedia.org/wiki/Null_Island).
```

  # `null` (the default): Unlimited

>   max_feature_count: null

Reviewer: Let's keep a real number here - `null` looks really weird tbh.    You can say unlimited if not set or set to `~` (null)?

Author: Well.. `~` in yaml is the same as `null`, and likely the part of the spec which users are not quite know.

What about setting this to a really high value instead?

```suggestion
  max_feature_count: 100_000
```

alternatively, we could change to a "float" (and truncate to an integer) and allow the following:

```suggestion
  max_feature_count: inf
```

Reviewer: nah, lets not introduce float, as that would be REALLY confusing.  A large number may work.
